src/assets/thirdPartyServices/snet/topic_analysis/index.js

```diff
@@ -121,15 +123,44 @@ export default class TopicAnalysisService extends React.Component {
   }
 
   submitAction() {
-    this.props.callApiCallback(this.state.serviceName, this.state.methodName, {
-      docs: this.state.dataset["docs"],
-      num_topics: this.state.dataset["num_topics"],
-      topic_divider: this.state.dataset["topic_divider"],
-      maxiter: this.state.dataset["maxiter"] === undefined ? "2" : this.state.dataset["maxiter"],
-      beta: this.state.dataset["beta"] === undefined ? "1" : this.state.dataset["beta"],
-    });
+    const {methodName} = this.state;
+    const methodDescriptor = TopicAnalysis[methodName];
+    const request = new methodDescriptor.requestType();
+
+    let docs = this.state.dataset["docs"]
+    let num_topics = this.state.dataset["num_topics"]
+    let topic_divider = this.state.dataset["topic_divider"]
+    let maxiter = this.state.dataset["maxiter"] === undefined ? "2" : this.state.dataset["maxiter"]
+    let beta = this.state.dataset["beta"] === undefined ? "1" : this.state.dataset["beta"]
+
+    request.setDocsList(docs);
+    request.setNumTopics(num_topics);
+    request.setTopicDivider(topic_divider);
+    request.setMaxiter(maxiter);
+    request.setBeta(beta);
+
+    const props = {
+      request,
+      onEnd: response => {
+        const { message, status, statusMessage } = response;
+        if (status !== 0) {
+          throw new Error(statusMessage);
+        }
+
+        // Check whether we need for the initial set of values
+        // this.state = this.getInitialState();
+
+        this.setState({
+          response: { status: "success", status: message.getStatus(), message: message.getMessage(), handle: message.getHandle() },
```

Comment:
It's better if we also set the initial state. This will help in the reset and run workflow.

Reply:
Duplicate status key defined.